dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/iter/RyaStatementCursorIterator.java

```diff
@@ -22,83 +40,83 @@
-			if (currentCursor.hasNext()) break;
-		}
-	}
-	
-	private boolean currentCursorIsValid() {
-		return (currentCursor != null) && currentCursor.hasNext();
-	}
-
-
-	public void setMaxResults(Long maxResults) {
-		this.maxResults = maxResults;
-	}
-
-	@Override
-	public void close() throws RyaDAOException {
-		// TODO don't know what to do here
-	}
-
-	@Override
-	public void remove() throws RyaDAOException {
-		next();
-	}
-
+    private static final Logger log = Logger.getLogger(RyaStatementCursorIterator.class);
+
+    private final DBCollection coll;
+    private final Iterator<DBObject> queryIterator;
+    private Iterator<DBObject> resultsIterator;
+    private final MongoDBStorageStrategy<RyaStatement> strategy;
+    private Long maxResults;
+    private final Authorizations auths;
+
+    public RyaStatementCursorIterator(final DBCollection coll, final Set<DBObject> queries, final MongoDBStorageStrategy<RyaStatement> strategy, final MongoDBRdfConfiguration conf) {
+        this.coll = coll;
+        this.queryIterator = queries.iterator();
+        this.strategy = strategy;
+        if (conf != null) {
+            this.auths = conf.getAuthorizations();
+        } else {
+            auths = MongoDbRdfConstants.ALL_AUTHORIZATIONS;
+        }
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (!currentCursorIsValid()) {
+            findNextValidCursor();
+        }
+        return currentCursorIsValid();
+    }
+
+    @Override
+    public RyaStatement next() {
+        if (!currentCursorIsValid()) {
+            findNextValidCursor();
+        }
+        if (currentCursorIsValid()) {
+            // convert to Rya Statement
+            final DBObject queryResult = resultsIterator.next();
+            final RyaStatement statement = strategy.deserializeDBObject(queryResult);
+            return statement;
+        }
+        return null;
+    }
+
+    private void findNextValidCursor() {
+        while (queryIterator.hasNext()){
+            final DBObject currentQuery = queryIterator.next();
+
+            // Executing redact aggregation to only return documents the user
+            // has access to.
+            final List<DBObject> pipeline = new ArrayList<>();
+            pipeline.add(new BasicDBObject("$match", currentQuery));
+            pipeline.addAll(AggregationUtil.createRedactPipeline(auths));
+            log.debug(pipeline);
+            final AggregationOutput output = coll.aggregate(pipeline);
+            resultsIterator = output.results().iterator();
+            if (resultsIterator.hasNext()) {
+                break;
+            }
+        }
+    }
+
+    private boolean currentCursorIsValid() {
+        return (resultsIterator != null) && resultsIterator.hasNext();
+    }
+
+
+    public void setMaxResults(final Long maxResults) {
+        this.maxResults = maxResults;
+    }
+
+    @Override
+    public void close() throws RyaDAOException {
+        // TODO don't know what to do here
```

Comment:
why are we implementing ClosableIterable when we ignore close()?.....not this problem though